component/loki/source/file/file_test.go

```diff
@@ -0,0 +1,61 @@
+package file
+
+import (
+	"context"
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/discovery"
+	"github.com/grafana/agent/pkg/flow/logging"
+	"github.com/grafana/loki/clients/pkg/promtail/api"
+	"github.com/prometheus/common/model"
+	"github.com/stretchr/testify/require"
+)
+
+func Test(t *testing.T) {
+	// Create opts for component
+	l, err := logging.New(os.Stderr, logging.DefaultOptions)
+	require.NoError(t, err)
+	dataPath, err := os.MkdirTemp("", "loki.source.file")
+	require.NoError(t, err)
+	defer os.RemoveAll(dataPath) // clean up
+
+	opts := component.Options{Logger: l, DataPath: dataPath}
+
+	f, err := os.CreateTemp(dataPath, "example")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer f.Close()
+	defer os.Remove(f.Name()) // clean up
```

Comment:
Are these in the right order? Shouldn't files normally be closed before they're removed?